tck/src/main/java/org/eclipse/microprofile/config/tck/broken/ConfigPropertiesZeroArgPL.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2016-2017 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.eclipse.microprofile.config.tck.broken;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.eclipse.microprofile.config.inject.ConfigProperties;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.ShouldThrowException;
+import org.jboss.arquillian.testng.Arquillian;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.testng.annotations.Test;
+
+/**
+ * Verify the bean supporting {@code ConfigProperties} must has a zero-arg constructor.
+ *
+ * @author <a href="mailto:emijiang6@googlemail.com">Emily Jiang</a>
+ */
+public class ConfigPropertiesZeroArgPL extends Arquillian {
```

Comment:
I would delete this whole test case. As I commented at the Config.getConfigProperties() method, we shouldn't verify that there's a zero-arg constructor. Implementations can choose whether they require it or can work around it, or even provide some custom features that use it.

Alternatively, if we want to enforce that in case an implementation requires a zero-arg constructor it throws `IllegalArgumentException`, we can just amend the test to pass only when either no exception is thrown or `IllegalArgumentException` is thrown.

Reply:
I am ok to remove this test if we have agreed on the javadoc change.

Reply:
removed